core/src/main/java/com/linecorp/armeria/client/http/HttpClientPipelineConfigurator.java

```diff
@@ -114,20 +116,13 @@
     private InetSocketAddress remoteAddress;
 
     HttpClientPipelineConfigurator(SessionProtocol sessionProtocol, SessionOptions options) {
-        switch (sessionProtocol) {
-        case HTTP:
-        case HTTPS:
+        if (sessionProtocol == HTTP || sessionProtocol == HTTPS) {
             httpPreference = HttpPreference.HTTP2_PREFERRED;
-            break;
-        case H1:
-        case H1C:
+        } else if (sessionProtocol == H1 || sessionProtocol == H1C) {
             httpPreference = HttpPreference.HTTP1_REQUIRED;
-            break;
-        case H2:
-        case H2C:
+        } else if (sessionProtocol == H2 || sessionProtocol == H2C) {
             httpPreference = HttpPreference.HTTP2_REQUIRED;
-            break;
-        default:
+        } else {
```

Comment:
> How does the SessionProtocol extendibility work? It looks like any new protocol would end up here right? Should we move protocol-related pipeline logic / fields all into the protocols themselves?

All HTTP-related stuff will end up here, but I was rather thinking about adding adding non-HTTP protocols, such as Redis, HBase and MySQL.

> This is good timing since I found grpc has switched to using the "grpc-exp" scheme, which is just an alias for H2. I'm not sure why this is a good idea, but since we're looking at extending SessionProtocol here, it seems like a simple enough use case that should be supported if possible.

That's interesting. If it's not something that affects the Netty pipeline but just is an alias, perhaps we could provide a way to add an alias to an existing SessionProtocol. What do you think?